### PR TITLE
Docs: Define Storybook's RFC process

### PR DESCRIPTION
**Pull request by vanessayuenn**
## What I did

- Direct `feature request` to opening a GitHub discussion instead of an issue (cc @shilman)
- Add a callout to opening RFCs in `issues/new` 
- Formalize our "RFC process" by documenting it.

## Outstanding tasks

- [ ] Find a home for the RFC doc. I just threw it under `contributing` for now, but @jonniebigodes @kylegach if you have thoughts on where this should live long term, please let me know. 
- [ ] Make sure the document is linked from appropriate places within the docs site

#### Maintainers

- [x] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [x] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: docs/contribute/RFC.md
Comment (JReinhold) [inline docs/contribute/RFC.md:33]: I think the point here about "accepted" != "prioritized" is very important. I can imagine that users will keep asking "any updates" often, I don't know if there's something to write here to mitigate that?

**Review by JReinhold — COMMENTED** [review]
This looks great!

File: docs/contribute/RFC.md
Comment (valentinpalkovic) [inline docs/contribute/RFC.md:29]:
```suggestion
Eventually, the team will decide whether the RFC is a candidate for inclusion in Storybook. On the other hand, an RFC may be rejected by the team after public discussion has settled and comments have been made summarizing the rationale for rejection. 
```

**Review by valentinpalkovic — APPROVED** [review]

File: docs/contribute/RFC.md
Comment (kylegach) [inline docs/contribute/RFC.md:15]: Minor, but the `###` heading level was skipped here.

File: docs/contribute/RFC.md
Comment (kylegach) [inline docs/contribute/RFC.md:1]: If this is intended to live on the docs site (I think it should!), then a couple additions are necessary:

1. Add the frontmatter and an H1 (`# RFC process`) to this file
2. Update the table of contents (`/docs/toc.js`) to add a link to this page in the correct section

If this is not intended to live on the docs site, then it should not be in the `/docs` directory. I think it best fits at the root of the repo. Or we could make a `/contributing` directory and move all relevant files in there (and update all references).

**Review by kylegach — DISMISSED** [review]
Thanks, Vanessa! The content looks great to me, but I left a comment about the file's location that must be addressed.

File: docs/contribute/RFC.md
Comment (jonniebigodes) [inline docs/contribute/RFC.md:19]: @vanessayuenn what do you think about adding some context here and inviting the author of the RFC or the person willing to implement it to the Storybook triage meeting to allow him to present his/her case? That would help us maintainers have some additional context on the RFC and get some additional context.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@vanessayuenn left one item for you to look into that I'm more than glad to follow up with. Let me know and we'll go from there.

File: docs/contribute/RFC.md
Comment (ndelangen) [reply docs/contribute/RFC.md:1]: > (I think it should!)

I agree, IMHO every single bit of non-code should be accessible from the website.

File: docs/contribute/RFC.md
Comment (ndelangen) [reply docs/contribute/RFC.md:3]: I thought RFC stood for "Request for Change"

File: docs/contribute/RFC.md
Comment (ndelangen) [reply docs/contribute/RFC.md:33]: Would this be the right place to talk about RFC stages / states?

Here's some draft:
- new
- discussion
- declined | accepted
- prioritized
- in progress
- alpha | beta 
- completed

File: docs/contribute/RFC.md
Comment (vanessayuenn) [reply docs/contribute/RFC.md:1]: Yes I intended for it to live on the docs site! Thanks for the pointers and I'll make the recommended adjustments.

File: docs/contribute/RFC.md
Comment (vanessayuenn) [reply docs/contribute/RFC.md:19]: That's a fantastic idea!

File: docs/contribute/RFC.md
Comment (vanessayuenn) [reply docs/contribute/RFC.md:33]: Thank you @ndelangen @JReinhold. The mental model here is that an RFC is a proposal, not a project tracking issue. Therefore it makes sense that the lifecycle of an RFC ends at accepted/rejected (see the section above). When the core team is ready to implement it, we can open a tracking issue and follow our typical processes. 

An "accepted" status means that "core team in principal agrees with the proposed changes", not "this is on our roadmap". I can certainly add something to make this point more clear.

**Review by jonniebigodes — APPROVED** [review]

**Comment by jonniebigodes** [comment]
Will merge once the checklist clears. Once again thanks for putting this together @vanessayuenn. 🙏

**Pull request merged**
